[doc] update readme

chewnoill opened the pull request:
updating readme

chewnoill committed:
[doc] update output content

chewnoill commented on README.md:
```diff
@@ -72,10 +72,44 @@ $ mix torch.gen.html Blog Post posts title:string body:text published_at:datetim
+* creating lib/phx1_6_web/views/post_view.ex
+* creating test/phx1_6_web/controllers/post_controller_test.exs
+* creating lib/phx1_6/blog/post.ex
+* creating priv/repo/migrations/20220201154756_create_posts.exs
```
`mix torch.gen.htm` creates a lot of files, besides templates, that seems like they are pretty important. Might be worth highlighting some of them.

cpjolicoeur replied:
Now that I see this, I'm not sure I want to keep this list in the README.  This list of files will surely change over time as Phoenix changes, and I don't want us to have to make sure it's always exhaustive and correct.   

If we do anything what about just showing the first line or two, then ... then the final line or two just to simulate what to expect?

```bash
* creating priv/templates/phx.gen.html/edit.html.heex
* creating priv/templates/phx.gen.html/form.html.heex
...<omitted for brevity>...
* injecting test/phx1_6/blog_test.exs
* injecting test/support/fixtures/blog_fixtures.ex
```

chewnoill commented on README.md:
```diff
@@ -72,10 +72,44 @@ $ mix torch.gen.html Blog Post posts title:string body:text published_at:datetim
+* creating lib/phx1_6/blog.ex
+* injecting lib/phx1_6/blog.ex
+* creating test/phx1_6/blog_test.exs
+* injecting test/phx1_6/blog_test.exs
```
whats the difference between `creating` and `injecting`?

cpjolicoeur replied:
That's the terminology Phx uses in it's generators.  Create is the actually creation of the file, inject is copying it into the correct directory location I believe.

chewnoill replied:
So files that aren't injected aren't placed onto the filesystem?
It must be doing some sort of post processing after creation, but I'm not sure what.

cpjolicoeur replied:
I dont know.  Never looked into it. It's not something that Torch is doing specifically.  That output is from the underlying phoenix generators that Torch calls.

chewnoill commented on README.md:
```diff
@@ -110,6 +149,32 @@ model and controller modules already exist, use the following command:
+* creating lib/phx1_6_web/templates/admin/post/new.html.heex
+* creating lib/phx1_6_web/templates/admin/post/show.html.heex
+* creating lib/phx1_6_web/views/admin/post_view.ex
+* creating test/phx1_6_web/controllers/admin/post_controller_test.exs
```
I'm not sure this is valuable, this is the same list as above but without the migrations.

cpjolicoeur replied:
Same comment here as above.  I'm not really keen on keeping a full list of files as the actual files and names probably are mostly unimportant to the end user in the context of reading this readme.

cpjolicoeur commented on README.md:
```diff
@@ -87,7 +121,12 @@ Ensure the following is added to your endpoint.ex:
     )
 
-  :fire: Torch generated html for Posts! :fire:
+Also don't forget to add a link to layouts/torch.html.
```
Can we add "if desired" to the end of this line as it's not a requirement.

chewnoill committed:
[chore] update cli copy

chewnoill committed:
[doc] cleanup readme

chewnoill committed:
[doc] add cli descriptions

This was helpful for me to list what options were being passed. Although
I think all of these options are specified and used by phx.gen.html.

cpjolicoeur commented on README.md:
```diff
@@ -66,16 +66,37 @@ about the supported options and format.
+* creating priv/templates/phx.gen.context/schema_access.ex
+* creating priv/templates/phx.gen.context/test_cases.exs
+* creating priv/templates/phx.gen.context/context_test.exs
+...<more output from phx.gen.html>...
```
Can you cut out more of these lines, perhaps keeping only 2 or 3 of the top lines like I mentioned before, please?

chewnoill committed:
[doc] remove extra lines
